cmd/soroban-rpc/internal/methods/get_transactions.go

```diff
@@ -134,92 +116,141 @@ func (h transactionsRPCHandler) getTransactionsByLedgerSequence(ctx context.Cont
 			limit = request.Pagination.Limit
 		}
 	}
+	return start, limit, nil
+}
 
-	// Iterate through each ledger and its transactions until limit or end range is reached.
-	// The latest ledger acts as the end ledger range for the request.
-	var txns []TransactionInfo
+// fetchLedgerData calls the meta table to fetch the corresponding ledger data.
+func (h transactionsRPCHandler) fetchLedgerData(ctx context.Context, ledgerSeq uint32) (xdr.LedgerCloseMeta, error) {
+	ledger, found, err := h.ledgerReader.GetLedger(ctx, ledgerSeq)
+	if err != nil {
+		return ledger, &jrpc2.Error{
+			Code:    jrpc2.InternalError,
+			Message: err.Error(),
+		}
+	} else if !found {
+		return ledger, &jrpc2.Error{
+			Code:    jrpc2.InvalidParams,
+			Message: fmt.Sprintf("ledger close meta not found: %d", ledgerSeq),
+		}
+	}
+	return ledger, nil
+}
+
+// processTransactionsInLedger cycles through all the transactions in a ledger.
+func (h transactionsRPCHandler) processTransactionsInLedger(ledger xdr.LedgerCloseMeta, start *toid.ID,
+	txns *[]TransactionInfo, limit uint,
+) (*toid.ID, error) {
 	cursor := toid.New(0, 0, 0)
-LedgerLoop:
-	for ledgerSeq := start.LedgerSequence; ledgerSeq <= int32(ledgerRange.LastLedger.Sequence); ledgerSeq++ {
-		// Get ledger close meta from db
-		ledger, found, err := h.ledgerReader.GetLedger(ctx, uint32(ledgerSeq))
-		if err != nil {
-			return GetTransactionsResponse{}, &jrpc2.Error{
+	reader, err := ingest.NewLedgerTransactionReaderFromLedgerCloseMeta(h.networkPassphrase, ledger)
+	if err != nil {
+		return cursor, &jrpc2.Error{
+			Code:    jrpc2.InternalError,
+			Message: err.Error(),
+		}
+	}
+
+	startTxIdx := 1
+	ledgerSeq := ledger.LedgerSequence()
+	if int32(ledgerSeq) == start.LedgerSequence {
+		startTxIdx = int(start.TransactionOrder)
+		if ierr := reader.Seek(startTxIdx - 1); ierr != nil && !errors.Is(ierr, io.EOF) {
```

Comment:
@Shaptic That is not necessary since this line will handle that:

```
for i := startTxIdx; i <= txCount; i++
```
The startTxIdx will be greater than txCount and it will exit automatically